### ISSUE TITLE
Change storage  Image rejection for mode to Image rejection for frequency

### ISSUE THREAD

**Issue by PA3CCE** [df8oe/mchf-github#233]
I like the new IQ balance adjustment,
now I can turn the image completely into the noise !

In the menu, storing of the IQ balance settings is provided
for the different modes LSB, USB etc.

In my case, I find no- or neglectable differences in
settings between the modes, but quite some difference
in the settings over the entire frequency range.
If confirmed, I should like to have the IQ balance in the
menu per mode, changed to settings per frequency
(Low-, medium-, High Frequency, or even better per band).

Jos (PA3CCE)

**Comment by df8oe** [comment]
Hello Jos,

this is a very good idea. When I first adjusted IQ with the new functions I had to increase power because I could not measure unwanted signal! And so I stated that there is a dependency to power, too. If you adjust it @0.5W and signal is completely gone, you switch to 5W and adjust that signal again is completely gone, then switch back to 0.5W you will see there is a slight misadjustment.

But we are talking about things which before nobody was able to see :)

We will think about improved structure, freuency dependency included. It is not high priority because now its working for the first time, but "good things can be improved" of course :)

vy 73
Andreas

**Comment by PA3CCE** [comment]
Well, I don't find the present situation real "good", when set at mid frequency,
the image rejction at low- and high-frequencies are rather "poor"

73 de
Jos

**Comment by df8oe** [comment]
...but _much_ better than all the years before :) So I call it 100% "good". Hundreds of users have lived without a working IQ-adjustment and nearly nobody has complained before. But now, if it works - it is _not good_ ??

Sorry - I cannot understand.

vy 73
Andreas

**Comment by PA3CCE** [comment]
A change from not working at all to 100% working for just one frequency 
is huge.
Working for the entire frequency range is  much better.
I think that many users just calibrate for one frequency and think " OK 
fine".
They  seem not to be aware what happens at  other frequencies.
Besides, I never heard you say before that the image rejection was not 
working at all,
or did I miss that ?

Best 73 de
Jos

Op 8-4-2016 om 11:06 schreef eeDF8OE:

**Comment by df8oe** [comment]
Hello  Jos,

haven't you read many posts in the past in Yahoo-NG that IQ-phase adjust does not work?? These posts go back to the time before I joined the project in May 2015... So the best adjustment in the past was not so good as if you adjust it with actual firmware and go to the band with the worst rejection. Because of it was not working properly before many users never have had adjusted their iq settings.

Additional problem may (again) be we reach edge of mcHF. The problem results of unsymmetrical capacities / inductivities at i and q LO signal line. I have already tested and they depend on many things like voltage of power supply, actual output etc. so it is possible that it is impossible to get all these dependencies included.

vy 73
Andreas

**Comment by PA3CCE** [comment]
Op 8-4-2016 om 16:06 schreef DF8OE:

> Hello Jos,
> 
> haven't you read many posts in the past in Yahoo-NG that IQ-phase 
> adjust does not work?? These posts go back to the time before I joined 
> the project in May 2015... So the best adjustment in the past was not 
> so good as if you adjust it with actual firmware and go to the band 
> with the worst rejection. Because of it was not working properly 
> before many users never have had adjusted their iq settings.
> 
> I agree, I also complained in the past.
>   With me I could adjust the amplitude, but the phase adjustment did not 
> react at all.
>   However, this is'nt "not working at all".
> To my opinion this is not to be used as a reference to declare the 
> present situation as "good".
> 
> Additional problem may (again) be we reach edge of mcHF. The problem 
> results of unsymmetrical capacities / inductivities at i and q LO 
> signal line. I have already tested and they depend on many things like 
> voltage of power supply, actual output etc. so it is possible that it 
> is impossible to get all these dependencies included.
> 
> Sure it will never be perfect, but what I proposed might be an 
> acceptable  approach.
> Do not underestimate the importance of the best image rejection in Tx- 
> as well in Rx-mode.

73 de
Jos

vy 73

**Comment by DD4WH** [comment]
Hi Jos, hi Andreas,
my impression is, we have all fully acknowledged that we need a smart solution for the dependencies of the phase adjustment in order to get the very best out of our hardware. But we should also be smart in using a slim solution, if it is feasible and appropriate. I am not sure whether a phase adjustment for every band would achieve this (and would be necessary), but we would probably need more data on that in order to be able to decide.

Could you two state the amount of sideband suppression that is lost when you adjust phase & amplitude perfectly for one band and then change to the band with the worst suppression? Is it in the range of 2-5dB or in the range of 10-20dB? And please also state whether your measurement is for TX or RX. 

Theoretically, (for TX) if we can adjust sideband suppression to an optimum of 55dB for one band and this gets worse for another band in an order of 6dB, well, then don´t let us talk about that again . . ., that´s still 49dB of suppression (10W -49dB equals 126µW, which is acceptable in my view).

For RX, with listening to DX signals and a local 100W station at 12kHz of your frequency (if you use the 12k IF), it could be nice to have 6dB more suppresion . . . however, the best hardware phasing receivers hardly ever achieve more than 45dB of sideband suppression (in the best band!).

Let us look into the HARDWARE signal I&Q chain, this is different for TX and RX:
RX: 
C68/C69 (22nF)--> they should be matched! I did that when I built the mcHF and I do not have such a big improvement with the new phase adjustment as others, maybe because of the matching, dont know!?  
C71/C73 --> must be 0R !
C26/C31 (470nF) --> maybe matching helps? Have not tried yet
C18/C19 (1µF) --> maybe matching helps? Have not tried yet
--> For RX, it should be easy to match the three pairs and have very good sideband suppression with the hardware alone

TX:
C7a/C8a (470nF)
C7/C8 (470nF)
C109-C112 (10µF)
C114-C120 (100µF)
C84-C87 (22nF) --> at least these must be matched!
--> There are not so many capacitors and no inductor in the I&Q signal chain. So, I do not see that it would not be possible to improve a lot on the hardware side by matching the capacitors in the I&Q TX line. 

Question 1: Maybe this also helps to decrease the frequency dependence of the phase adjustment?

Question 2: Is it necessary to distinguish between phase adjustment for LSB & USB, AM & FM?
--> At least theoretically, I do not see any reason for this: All the demodulation steps occur in software. So, if we adjust I& Q phase and amplitude before demodulation (as is the case in the moment), this adjustment should be the same and should hold for ALL demodulation modes.
(It would be good, however, if that could be supported by your observations!) 

This could free up EEPROM and menu space for frequency dependent adjustments!? --> Just a thought . . .

added later:
As comparisons of the mcHF and the KX3 seem fashionable at the moment (;-)), it is worth noting that the KX3 also has separate IQ phase adjustments entries for EACH band.

73 de Frank DD4WH

**Comment by PA3CCE** [comment]
Hi Frank,

First of all I must tell you that at this moment I cannot perform any 
test, since
today my mchf stopt working. Something is wrong in my LO.
At times the frequency display is in red color and the spetrum display is
interrupted in abt 1 sec rate.
When I change frequency, the spectrum display stops inmediately.
Could not find the cause so far.

I also can tell you that I have made all modifications as they are in 
the circuits
of IZ6MAF, so all unnecessary capacitors are shorted etc.
I did the tests in Rx mode only.
Further more I had matched the appropriate components in the demod circuit.

I can give you the adjustment figures  below :

Freq                   LSB                            USB
                     ampl/phase            ampl/phase

  3MHz                 5/2                         5/1
  5MHz               11/5                      11/4
  7MHz               17/8                      18/7
10MHz           19/14                    20/13
14MHz           30/22                    28/22
21MHz           50/38                    54/37
24MHz            58/52                   58/50
28MHz            63/65                   63/65

These are the settings I found in the menu (all positive).
You see that the difference between the demod modes are neglectable,
this was the reason I did not test the other modes.
You also can see large differences per frequency, in phase as well in 
amplitude.
I did not write down the differences in image rejection but, since you 
know the
change per step, you can have an idea.

As soon as I have mcHF working again I can profide you more info.
I hope this helps.

best 73 de
Jos

Op 8-4-2016 om 19:43 schreef DD4WH:

**Comment by DD4WH** [comment]
Hi Jos,

thanks a lot for your data. That helps, also it would be very good to also have settings translated into dB of suppression ;-) I altered the setting of the finetuning a few versions of firmware ago, so a difference of 5 to 63 is not so big anymore as it was in former times ;-)

I had the same problem with red digits and it returns from time to time. Check soldering of the Si570 and around that, very likely that the problem is there, but Andreas probably knows more.
From your data, I would conclude, we can stop distinguishing between modes in the phase and amplitude adjustment! Also, if you have matched capacitors in the Rx chain, it would have to be concluded that this does not help against the frequency dependance of the adjustments! Very interesting and valuable results. I will think about that and wait for more info from other users.

Thanks a lot for your info and good luck in fixing your rig! If it is ok, I would appreciate very much more info, eg. from TX mode and other measurements!

73 de Frank

**Comment by df8oe** [comment]
Hello Jos,

that is very interesting! My settings differ only +/- 2 (both phase and gain) between optimum and worst band.

Why does yours differ so much? Now I understnad your opinion - but I think, you do have a hardware problem. All of my 4 mcHFs differ only +/- 2...

y 73
Andreas

**Comment by PA3CCE** [comment]
Hello Andreas

+/- 2 difference over the entire frequency range, are you sure ?
That is almost ideal, I can hardly believe that.
I hope my mcHF will be operable soon, I ordered a new RF board,
the old one is, due to many experiments, worn out.

best 73 de
Jos

  8-4-2016 om 23:58 schreef DF8OE:

**Comment by df8oe** [comment]
Yes. I have one mcHF built as RX-only and three mcHFs built as transceivers.

but I have made a modification on all of these which is not documented yet - maybe that is the reason??!!

I kicked off drivers U12 and U13 on rf-board and swapped R22...R25 with 0R. Drivers are producing more phase shifting at higher frequencies and in my eyes are not neccessary (and maybe they do disturb more than they are useful). But I have not made intensive measuring. Of course this can be the reason for the great differences between your mcHF and mine!

Must be investigated... But I am only one person and have to make business, too , so I do not have much time for all of this. McHF takes already big part of my time :)

vy 73
Andreas

**Comment by DD4WH** [comment]
Very impressive differences! Hmmm . . .

Andreas, is that in RX or TX or both? Are the differences between modes also so small?

Would love to try your mod, did you solder out the SN74LVC2G17s or did you just solder two wires on top of them to bridge the buffers?

73 de Frank

P.S.: Unfortunately, I am not able to measure myself, because I have no stable RF generator with such a high drive . . .

**Comment by df8oe** [comment]
You must desolder little drivers and connect in/out (2x at each driver) with wire. My settings are in a small range at RX and TX.

vy 73
Andreas

**Comment by PA3CCE** [comment]
@ Andreas, Frank,

OK Andreas that is most interesting information, thank you.
As a matter of fact, the new RF board I ordered is of version 0.5.
In this version these drivers are rejected indeed and the traces
of the LO are matched for impedance, as described by Chris.
As soon I have the results, I will let you both know.

Have a nice weekend,
Jos

Op 9-4-2016 om 14:18 schreef DF8OE:

**Comment by df8oe** [comment]
Hello Jos,

thank you for this information. That will be an explanation... Additional I have used 7474 manufactured by "Potatoe Chips" which has anupper working frequency of 1GHz. This will become eccessary when you will add the rf-mod-pcb so that 6m and 4m are available, too. Maybe timing at higher frequencies is more precise like on normal logic IC...

Your feedback after getting new pcbs is welcome!

Have a nice weekend, too
Andreas

**Comment by PA3CCE** [comment]
Hi Andreas

Thanks again, I must have some "Potatoe Chips" circuits somewhere too, 
so I will see.
I hope and expect theat I can keep using the UI version 0.4, for I just 
ordered the RF board.

Cheers
Jos

Op 9-4-2016 om 16:11 schreef DF8OE:

**Comment by DD4WH** [comment]
Very interesting, these 7474 chips! Did you also use the faster potatoe version of the 3253 as substitutes for the 3253 Muxer? The potatoe version seems to have much larger on resistance (17-22ohms vs 3-6ohms), could potentially be a problem! 

73 de Frank

**Comment by DD4WH** [comment]
I could now reactivate my Dipmeter and an RF amp and an attenuator, so I had an RF source at hand which produced  a huge RF signal with -17dBm. Here are my settings for optimum IQ amplitude and phase adjustment setting for specific frequencies:
(frequency, (ampl., phase for LSB), (ampl., phase for USB))

2.5MHz  -14/-20 LSB  -14/-20 USB
4MHz  -13/-19 LSB  -12/-19 USB
8MHz  -13/-20 LSB  -13/-20 USB
12MHz  -20/-34 LSB  -23/-23 USB
16MHz  -16/-30 LSB  -16/-27 USB
21MHz  -24/-31 LSB  -25/-29 USB
24MHz  -34/-49 LSB  -35/-44 USB
28MHz  -37/-55 LSB  -38/-46 USB

side observation: at 24MHz and 28MHz the nulling was dependent on where I put my fingers on the mcHF (which has no shielding or metal case in my mcHF)!!! That could also be one of the reasons for your varying results, Jos!?

My conclusions:
1.) we do not need to distinguish between LSB, USB, AM, SAM, FM in the amplitude & phase adjustments --> software can be simplified
2.) the adjustment is more or less frequency dependent (depends on your build/your measurement situation) --> whether we need a band-specific possibility to adjust, is not clear yet, because in my case (see above), it would really be overkill . . . but in others (Jos!) it could be nice to have.
3.) The frequency dependance is probably 100% the result of the buffer ICs U9, U12, and U13 (if you compare Andreas´ results with Jos´ and my results). The buffer ICs U9, U12, U13 and corresponding Rs R19, R22-R25 should be eliminated as a suggested modification.

Comments and more measurement data very welcome!

73 de Frank

**Comment by df8oe** [comment]
I do not have swapped mixers (due to quite different Ron what will be a problem). But I have tested nd they are usable up to 100MHz - they will work with 6m and 4m od, too.

vy73
Andreas

**Comment by PA3CCE** [comment]
Thanks, see below.
Jos

Op 10-4-2016 om 13:09 schreef DD4WH:

**Comment by DD4WH** [comment]
I just soldered out U12 and U13 and put in 0R for R22-R25. So these buffers are now out of the I&Q path:
Did the same measurements again for RX:

2.5MHz  -10/-22 LSB     -7/-22 USB
4MHz  -11/-24 LSB     -4/-23 USB
8MHz -3/-29 LSB     2/-28 USB
12MHz 2/-33 LSB     5/-33 USB
16MHz 11/-31 LSB     14/-31 USB
21MHz 4/-44 LSB     7/-43 USB
24MHz 5/-72 LSB     7/-72 USB
28MHz 33/-56 LSB     35/-56 USB

So, what would you conclude from this?
Comments welcome! ;-)
73 de Frank

**Comment by DD4WH** [comment]
I would conclude that my measurements at 24/28MHz probably are not reliable, because the (very strong) signal of -17dBm seems to enter the mcHF over additional paths . . .
However the other measurements seem to indicate the following:

@2.) we need a frequency dependent I&Q amplitude and phase adjustment. I fully agree with you, Jos, now with those measurements. A difference in adjustment of +-3 can result in a difference in mirror rejection of >20dB (not measured, but a sophisticated guess by me). Sometimes +2 steps turns absolute silence to a rather loud annoying heterodyne.

@3.) my hypothesis must be rejected: the buffers are not the problem! New hypothesis: the 74LCX74M (U11) is the problem, that chip seems to have a huge frequency dependence in handling amplitudes and phase. I will order the "wonder chip" (7474) by Potatoe chips in order to validate that.
Andreas, could you confirm, it is this one, please:
http://www.ebay.com/itm/7474-G-Series-GHz-TTL-CMOS-logic-IC-14pin-SOIC-QTY-1-/330551715157

73 de Frank

**Comment by df8oe** [comment]
YES - that is the one I am using.

vy 73
Andreas

**Comment by PA3CCE** [comment]
Op 10-4-2016 om 14:39 schreef DD4WH:

> I would conclude that my measurements at 24/28MHz probably are not 
> reliable, because the (very strong) signal of -17dBm seems to enter 
> the mcHF over additional paths . . .
> 
> Might be, but see my findings also.
> Can't you make a simple attenuator of say 20dB or more ?
> OK, you are using a dipmeter, may radiate a lot :(
> 1 turn loop on a long cable ?

However the other measurements seem to indicate the following:

> @2 https://github.com/2.) we need a frequency dependent I&Q 
> amplitude and phase adjustment. I fully agree with you, Jos, now with 
> those measurements. A difference in adjustment of +-3 can result in a 
> difference in mirror rejection of >20dB (not measured, but a 
> sophisticated guess by me). Sometimes +2 steps turns absolute silence 
> to a rather loud annoying heterodyne.
> 
> This matches my findings, 1 step makes a difference of no tone to a 
> noticeable tone.
> I also could make use of the S-meter, no signal was approx. S3 of noise.
> Signal input was S9 + 30 dB or so.
> I could calculate the approx. rejection  1 S-unit = 6dB
> 
> @3 https://github.com/3.) my hypothesis must be rejected: the 
> buffers are not the problem!
> 
> Looks worse than better, but as said there are probable more variables, 
> it's tricky.

New hypothesis: the 74LCX74M (U11) is the problem, that chip seems to 
have a huge frequency dependence in handling amplitudes and phase. I 
will order the "wonder chip" (7474) by Potatoe chips in order to 
validate that.

**Comment by DD4WH** [comment]
After some time for thinking, I have some new findings:

1.) With the buffers in my mcHF, there were no differences between LSB/USB. After taking out the buffer ICs, measuring, and after that additionally resoldering my solder bridges and measuring everything again, there are still significant differences between USB & LSB, but only in the amplitude adjustments, not in the phase adjustment! Thus, the buffers U12/U13 seem to be useful for eliminating frequency effects on the IQ amplitude differences. So, we do need software diiferentiation between LSB/USB adjustment, because RF board 0.5 has no more buffers.

2.) & 3.) Even IF the potatoe chip 7474 eliminates frequency dependancy, probably not every user will order it and build it in. So do we want a software adjustment? If yes, I would suggest: 12 different adjustments with the following low corner frequencies which I have chosen for ham radio use AND SWL in mind:
100kHz - 2200m
450kHz - 630m
1.75MHz - 160m
3MHz - 80m
5.5MHz - 40m
8MHz - 30m
11MHz - 20m
14.5MHz - BC 19m
16MHz - 17m
19MHz - 15m
22MHz - 12m
25MHz - 10m 

12 adjustments for amplitude & phase for USB/LSB separately means 48 adjustments (and also 48 EEPROM memory places)!

73 de Frank

**Comment by df8oe** [comment]
After we got a working IQ-adjustment -and first at this point - we consider that there is huge hardware problem sleeping in design of PCB. Because of IQ-lines are not designed @50 ohms there are frequency dependencies which never were recognized before. Capacities of mixer inputs and output impedance of drivers (or of 7474) are involved, too.

I think before we start a huge storing orgy we first have to accumulate informations. Maybe it is possible to find a formula how better scalings can be calculated. But therefore we need much more samples of actually best IQ adjustments...

vy 73
Andreas

**Comment by DD4WH** [comment]
Yes, you are right, the new software now makes the hardest hardware tests possible for the first time!

And the few results are still not leading into one direction and hardware effects are complex. 

So let us do more tests and gather more data under the diverse configurations.

I am really curious whether the 7474 chip from Potatoe chips will solve all that . . .

73 de Frank

**Comment by PA3CCE** [comment]
@Andreas, Frank

Just my penny :

1)   I do NOT see the necessity of making difference in adjustment in  
demodulation modes.
        dependency of other factors are many  many times  bigger.
        Besides, I have never seen this in other projects either.

2) I do NOT believe in one  satisfactory adjustment for all frequencies 
(and on all existing mcHF variants) ,
     there are too many dependencies  involved ( trace impedances, stray 
capacitances, stray inductances,
     component variations, etc. etc.) .

Best compromise :
Be realistic and make the best (simplest) solution for everyone, provide 
for an adjustment for each
frequency band (maybe more for the highest frequencies), NOT regarding 
modulation mode.

Best 73
Jos

Op 10-4-2016 om 16:55 schreef DF8OE:

**Comment by df8oe** [comment]
It is not trivial to implement so much storing (especially for users who don't have fitted serial EEPROM because in actual virtual EEPROM there is a cut at 383 words).

So let us "be patient" and find a solution.

Again:
all these problems came up a few days ago when IQ adjustment works as fine as is is now....

I don't think it is a problem in waiting some more days (or weeks) for finding best compromise - problem exists from the birth of mcHF on!!!!

We are pointed on this and working on it to get best solution.

vy 73
Andreas

**Comment by PA3CCE** [comment]
It's your choise.
How many stores do you have now, for all modulation modes !
When you find a better solution you can change any time.

Jos

Op 10-4-2016 om 18:49 schreef DF8OE:

**Comment by df8oe** [comment]
We don't have enough at the moment for users without serial EEPROM.

At the moment we do have 10 storage places. If we orientate in frequency only for HAM usage we need 12 and when we add some for SWLs we need 16.

vy 73
Andreas

**Comment by PA3CCE** [comment]
how do they cope now with all these modulation mode stores ?
No argument, sorry.

Op 10-4-2016 om 19:00 schreef DF8OE:

**Comment by DD4WH** [comment]
Keep calm, guys ;-) Rome has not been built in a single day . . .

We have all understood the problem and the arguments now and now it is time to stop talking, to measure again, to think hard again for the best optimization (because we are talking about an already superb mirror rejection here --> in the early days of the SDR cube 25dB was called "very usable" and we have >60dB when propery adjusted!!!), and then to start programming with an economic view on the EEPROM storage room :-)

Have fun!

73 de Frank

**Comment by PA3CCE** [comment]
OK Frank,

No problems here, I just like real arguments ;)
Will wait and see patiently what happens.
Will let you guys know when I have more info.

Jos

Op 10-4-2016 om 19:23 schreef DD4WH:

**Comment by DG9BFC** [comment]
what about storing some set points and the mcu calculates the points in between??

say you have a point of phase /gain at 1 megs of 20/30 and another at 6 megs with 25/35 ...

and you find out that the gain and phase is almost linear over that range (means at 2 megs you need 21/31 and at 3 megs 22/32) ... then you just store a few rough set points ... and rest is done in a calculation step inside the mcu

if at lower or higher bands the phase and gain correction is not that linear ... then just spread in another set point .... guessing that with a few points an almost ideal curve can be stored ... without the need of a too high number of set points

**Comment by df8oe** [comment]
That's the idea we have had, too! But there exist many possible modifications (e.g. swap 7474 with "potatoe chip", leave uffers like in rev. 0.5 is, different mixers...) All of these will change frequency  dependency of IQ lines. So we will start investigations next weeks and hope as many users as possible will participate. We will offer a spreadsheet where every users fills in his personal IQ settings for many frequencies and modes, added his hardware he uses. Then we hope we can see if a function exist and how to detect if there are different possibilities.

y 73
Andreas

**Comment by HB9LFU** [comment]
Hello Frank / Andreas and others,

thanks a lot Frank for sending me the link to the message about signal quality and using the tune button. With a real signal the spectrum is compeletely different and does not show these spurs.

However, there was one left close to the desired signal. But I could reduce it drastically with the IQ menu settings. On 80m -50 was ideal. On 10m -50 was fully indadequate, I had to crank the value up to almost +99. The value for 60m was also ok on 80 and 40m. Also 30 and 20m were fairly close, the same holds true for 17 and 15, as well as 12 and 10m. 

vy73s
Roger

**Comment by DD4WH** [comment]
Hello Roger,

nice to hear that the problem is solved now! The TUNE function is very
nice, but it is not good for checking signal quality ;-)

And also the phase adjustment does solve many "problems". However, we will
collect more data on that and Andreas is preparing a spreadsheet for
structured data collection and it would be very nice if you (and many
others!) could participate. It will be prepared in the next weeks and
announced in the Sulingen forum.

Have fun with the mcHF!

73 de Frank DD4WH

2016-04-14 20:43 GMT+02:00 HB9LFU notifications@github.com:

> Hello Frank / Andreas and others,
> 
> thanks a lot Frank for sending me the link to the message about signal
> quality and using the tune button. With a real signal the spectrum is
> compeletely different and does not show these spurs.
> 
> However, there was one left close the desired signal. But I could reduce
> it drastically with the IQ menu settings. On 80m -50 was ideal. On 10m -50
> was fully indadequate, I had to crank the value up to almost +99. The value
> for 60m was also ok on 80 and 40m. Also 30 and 20m were fairly close, the
> same holds true for 17 and 15, as well as 12 and 10m.
> 
> vy73s
> Roger
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/df8oe/mchf-github/issues/233#issuecomment-210094072

## 

---

Frank - DD4WH
QTH - Dresden, Germany
Rx/Ant remote - Perseus SDR & K9AY lowband antenna
Rx/Ant local - Teensy SDR standalone homebrew ugly style Rx

## & AAA-1 Bulgarian active Antenna with double loop

**Comment by HB9LFU** [comment]
Hello Andreas,

this eeprom is so cheap and it does not take an hour to open the case and solder an eeprom to the ui-board that this part should become a mandatory thing. 

vy73s
Roger

**Comment by PA3CCE** [comment]
Hello Andreas
Please find below the I/Q balance settings I found for my new RF board version 0.5 :

# Rx I/Q adjustment : (mirror inaudible)

Conditions :
Vi = -50dBm @ 50E
Rx Preamp = 5dB pad + BGA103+ 
LO divide by 4 (U11) = Potatochips PO74G74A
mcHF UI board version 0.4
mcHF RF board version 0.5
All IZ6MAF modifications implemented.
Firmware 1.1.0 / BL 1.0 (May 20.. 21:06:47 EST)

3.65MHz 
Amplitude :. ...USB..10.....LSB  10
Phase : ..........USB  -15....LSB -14

 5.250MHz
Amplitude :.....USB..20.....LSB...18
Phase :..........USB.-14.....LSB .-10

 7.10MHz
Amplitude :......USB..22.....LSB..18
Phase : ...........USB.-11.....LSB...-8

10.10MHz
Amplitude :.......USB..22....LSB..22
Phase :.............USB..-5.....LSB..-5

14.1 MHz
Amplitude :......USB..35.....LSB.. 32
Phase :............USB...1......LSB....1

18.69MHz
Amplitude :.....USB..42.....LSB..38
Phase :...........USB..15.....LSB..17

21.2 MHz
Amplitude :.....USB..54.....LSB..53 
Phase : ..........USB..16.....LSB..17

24.89 MHz
Amplitude :.....USB..63.....LSB..61 
Phase : ..........USB..22.....LSB..22

28.500 MHz
Amplitude : .....USB..69.....LSB..69
Phase ;............USB..52.....LSB..52

Best 73
Jos

**Comment by db4ple** [comment]
Hi,

your measurements are great. It seems to me that there is almost no difference between USB and LSB settings. For those where there is a difference, can you live with the  average between both settings? Would allow us to use just one set of settings. 

What I can see is there might be some formula to calculate the variation across the frequency from the low frequency  and high frequency value. Catch is, we would need more data to do validate that.  

That would limit the number of calibration steps in order to get to a better working mcHF. And we could reuse the existing EEPROM Layout.

73
 Danilo

**Comment by PA3CCE** [comment]
Hi Danilo,
Sure I can live with the average between the settings for USB and LSB, these diferences
are marginal.
It would be nice if you are able to implement a formula for the variations by frequency, it would be a big improvement in mcHF's performance.
Of course we need more information from other users to verify.

vy 73
Jos

**Comment by HB9LFU** [comment]
Hello Danilo,

is there not enough storage space to have a per band adjustment of these settings?

vy73s
Roger

**Comment by df8oe** [comment]
There is. But we want to keep adjustment menu and procedure as easy as possible and would prefer not to dig through a huge number of settings.

vy 73
Andreas

**Comment by HB9LFU** [comment]
Hello Andreas,

then why not adjust 80-60-40, 30-20,17-15 and 12-10 together. This would simplify the process considerably. I do not believe it is a good idea to calculate some sort of curve for the whole range and for all the different sets of mcHFs with varying modification levels. 

vy73s
Roger

**Comment by PA3CCE** [comment]
Hi all

When a formula alone does not work, a combination of both band grouping and a formula might come close to the ideal.

73 de
Jos

**Comment by df8oe** [comment]
We are thinking about all this possibilities. There are BC bands, too and we must find the best solution for all users. We must collect more data (TX, too) for 0.4 and 0.5 PCB and different modification stages so we can find final solution.

vy 73
Andreas

**Comment by PA3CCE** [comment]
I/Q adjustment for the HAM bands is a way of band grouping for the BC bands !

73 de
Jos

**Comment by db4ple** [comment]
Maybe to clarify: Even a formula approach will need some device specific calibration values, in order to to calculate the intermediate values for  the various frequencies.

**Comment by PA3CCE** [comment]
Hi Danilo,  Andreas and others,

Today I compared I/Q balance results from my old version .04 RF board to the last results from the new .05 board.

Amplitude :  5 to 63  (old board)  vs  10 to 69 (new board).
This is a variation of +58 (old) vs +59 (new).

Phase     :    2 to 65 (old board) vs  -14 to 52 (new Board) .
This is a variation of +63 (old) vs +66 (new).

I find this similiarity remarkable.
Take into account that the LO has changed importantly :

1) U11 74LCX74M (old), PO74G74A (new)
2) No buffers 74LVC1G17 and 74LVC2G17 (new)
3) Different LO layout (traces)
4) I used an other specimen of Si570 in version 0.5

So, it looks like the variation over the frequency range is more consistent then I expected.
This gives me confidence in you finding a working solution for all users.

I will give you my results for the Tx part, as soon as available.

Best 73 de
Jos

**Comment by HB9LFU** [comment]
Hello Danilo,
as promised here my IQ correction data for 80 to 10m. As I used cw with a key I could only vary USB TX IQ Balance:
80m=  21
60m=  42
40m=  27
30m= -52
20m= -61
17m= -67
15m= -75
12m= -80
10m= -99

regards
Roger

**Comment by df8oe** [comment]
...and here my data for the same working condition:
USB TX IQ Balance:
160m: -38
80m:  -37
60m: -37
40m: -38
30m: -34
20m: -26
17m: -12
15m: -10
12m: -6
10m: 1
6m: 20 (measured directly after mixer because filters are not ready yet)

As you see there is absolutely nothing similar...

vy 73
Andreas

**Comment by HB9LFU** [comment]
Yes, quite different Andreas!

vy73s
Roger

**Comment by df8oe** [comment]
I have four mcHF with a working TX. All four have nearly the same modifications. But all of them have quite different settings. Not only the values: one is going + and  - swapped. All have different slopes, two have a "glitch" wher e setting does do a jump (not at the same frequency).

_Very difficult_ to handle...

vy73
Andreas

**Comment by db4ple** [comment]
Let me put it in other words: The current approach is not worth being kept. Given the amount of change of the frequency range shown in all measurements a single value makes absolutely no sense.
I am wondering if we can confirm that settings for RX AM, USB, LSB for the same frequency are more or less equal. That would lessen the amount of values being required.
What worries me are the glitches. That kind of thing breaks almost all interpolation strategies since a glitch of that amount will make very strange values all the way to the next known value.
However, any approach which has some sort of per frequency values is better than the current one.

For the storage of the configuration values (as Andreas said) we have to extend our approach to have "EEPROM Only Values". I.e. values which are more or less directly written to EEPROM and always read from there (which is fine for low frequency write value, also  the PA adjust/coupling values fall into that category if you thing of it. We should be able to do that, its not too difficult to do.

73
Danilo

**Comment by HB9LFU** [comment]
Hi Danilo,

I do only cw so I cannot confirm that the other modes have the same values on the same band. But from what I heard from others does it not make a huge difference. So it could be ignored.

What I find important is to not  base future developments on those who have not yet installed the external eeprom. This part is cheap and does not take half an hour to install. 

Yes, with the glitches a per band solution seems inevitable. 

vy73s
Roger

**Comment by df8oe** [comment]
Hi Roger,

virtual EEPROM is an important fallback for users of serial EEPROM, so it MUST work. The option to backup and restore a set of settings is an important goal of the "EEPROM modification" and a _MUST_. But if we can expand virtual  EEPROM for backup purposes it can be used for those who do not have serial EEPROM without any further changes.

vy 73
Andreas

**Comment by HB9LFU** [comment]
Hello Andreas,

you have pushed through the touch screen use, now I think it is time that you push through the use of the external eeprom.
I am no friend of touch screens but had to accept it. 

vy73s
Roger

**Comment by df8oe** [comment]
Hi Roger,

programming work for creating storage of many IQ datasets is identical if use of serial EEPROM is mandatory or not...

We do not want to loose capabilityof backing up serial EEPROM to virtual EEPROM so we must expand it, too, regardless if serial EEPROM is available or not.

vy 73
Andreas

**Comment by HB9LFU** [comment]
Andreas, is there enough storage space in the cpu to keep  i-q correction data of every single ham band?

vy73s
Roger

**Comment by df8oe** [comment]
Yes, it is. There is room for 16KB data - enough for hundreds of datapoints AND hundred memory places for storing stations/modes/names/frequencies...

vy 73
Andreas

**Comment by arekh** [comment]
Hi, I'd just like refresh the topic with new hopes (since we have a lot of memory) :), this feature will be very useful and it seems that not only for me

73!
Arek

**Comment by IZ6MAF** [comment]
For me is to hard to understand, but i think that this article can help the programmers to solve the question
[aiqben.pdf](https://github.com/df8oe/mchf-github/files/468206/aiqben.pdf)

**Comment by DD4WH** [comment]
Hi!
thanks a lot for that paper! Very interesting, although the math is far too advanced for me ;-) But it´s nice to know that "automatic" IQ correction is possible (at the expense of high intensity calculations)! 

The only precondition would be that we would have a quad core processor with 2GHz or more in the mcHF ;-).

To be honest, a real time continous FFT/DFT with 4096 taps and subsequent complex calculations are absolutely too much for our poor processor . . . That is all possible if you write programs for a PC, but for us, it´s very hard optimizing for that little processor we have. Danilo, for example, has been spending hours and days (weeks?) in optimizing the code to be more efficient on the STM32F4, a thing not necessary when you program for a PC . . .

A very interesting method, but -in my opinion- unfortunately impossible to implement on the mcHF, that´s a pity . . . 

However: Please keep on posting links to interesting papers like this (even if we do not have the faintest chance to implement that on the mcHF) ! Maybe sometime we can take some of the brilliant ideas available on the SDR paper market and implement them.

All the best 73s, Frank DD4WH

**Comment by shkmr** [comment]
Hi, I made a plot from data shown here plus my own measurement.

http://shkmr.github.io/mchf/rxIQ.html

At first, I thought this mismatch is comming from different loading of Q1 and Q1n of 74AC74 and internal encoder of 74CBT3253. If it is the case v0.5 board should show same tendency, but it is not.

Anyway, I think linear interpolation is likely possible, by considering receiver performance in normal operation is not very sensitive to these settings....

**Comment by df8oe** [comment]
It would be very interesting if we can calculate settings in a way. From starting this issue to now there are some changes in firmware to get similar settings at one frequency reagarding all modes. Hopefulla these struggles are finished now and we can reduce settings to a simple frequency dependency.

Unfortunately in the actual build there is a new problem we do not have fixed until now. TX IQ settings for AM and FM (?) is ignored :( That is due to a refactoring of signal path. We know the "bad commit" and we will fix it.

Because of no one has paid attention with which build the data you used is gathered it maybe not comparable. If I guess "the best case" all curves are going upward (or downward). To work on this we _must_:

1) fix the AM/FM issue
2) start new data collection using all same firmware builds

I am sure frequency dependency is related to:
- type of 7474
- output buffers on board 0.4
- type of switches
- PCB layout

I for myself use flip flop and switches produced by "potatoe chips" which are usable up to ~ 1GHz so that we can use mcHF up to 4m...

73 de
Andreas

**Comment by df8oe** [comment]
1) is finished - we can start 2)...

73 de
Andreas

**Comment by db4ple** [comment]
Hi, 
I propose to use a common template to record all relevant in order to be able to see two things:
- The mcHF relationship of values over frequency. 
- Dependency from mcHF hardware aspects.
  Also the measurement setup should be documented. 
  We should also standardized the modes / output power being used. I suggest to take readings
  on at least 0.5W and 5W settings. Receivers should be reasonably calibrated in output power.
  I don't know if all users should use two powers, but I reckon, it should not be too hard to cycle through the different power settings, since this can be done on-the-fly.
  The good news is, we need only one mode (e.g. LSB) since values for all of the modes should be virtually identical.

The template can be OpenOffice and/or text (generate from the OpenOffice doc). This should suite all of us. 

Now: what needs to be taken into account as hardware items (anything to add to Andreas' list?)
Any other things to put on the template?

**Comment by df8oe** [comment]
- mcHF version: (0.4 / 0.5)
- type/manufacturer of CBT mixer
- type / manufacturer of divider
- buffers mounted (only 0.4)
- T4 (or surrounding) modded (if: what kind of mod)

Then we can confirm if settings are nearly idenrical on all modes and report, if not.
Next we must get optimal settings for RX and for TX at 0.5 and 5W output on:
- 1900
- 3700
- 7100
- 10130
- 14100
- 18100
- 21300
- 24950
- 28500

Please correct what is missing. I will set up a text file which easily can be used and exported as csv later.

**Comment by DG9BFC** [comment]
a list with all mods done ... cause that may influence output power

it will not influence phase and gain corrections ... but gain settings 
for the two different power levels

dg9bfc sigi

Am 03.10.2016 um 15:58 schrieb DF8OE:

**Comment by df8oe** [comment]
Ideally it is not a function of output power but a function of input amplitude 0°/90°/⁸0°/270° . But I am not sure that we can gather enough data if we complicate measure procedure.

So hopefully I think we get much more data if we use output power and are able to get IQsetting = function (power) _AND_ IQsetting = function (frequency) :)

**Comment by s53dz** [comment]
I just want to add some my old measurements of RX mirror to show the extent of spread.
The given worst case figure of S8 is quite disturbing compared to that what is possible to compensate for on each band: <S2.
The HW rev is 04, LO buffers were in.
Prior to these measurements I lowered the image signal by filtering /blocking the carrier directly at the mid tap of T1 using a capacitor.
https://uk.groups.yahoo.com/neo/groups/M0NKA-mcHF/files/S53DZ/settings/

73 Bojan S53DZ

**Comment by db4ple** [comment]
I have been looking at the data in the diagram (thanks shkmr). I think for now we can get away with a simple linear interpolation for the RX. Yes, I know, some curves are not strictly linear especially in the higher frequency area. What I want to do is to improve the situation with a very simple approach:

Calibrate at 3.5 Mhz -> store in former entry for LSB (now IQAdjustLow) 
Calibrate at 28Mhz -> store in former entry for USB (now IQAdjustHigh)
Calculate value for frequency with:   IQAdjustFreq =   (IQAdjustHigh-IQAdjustLow)/(28-3.5) * (f - 3.5) + IQAdustLow

That should give better values over a larger range of frequencies as today and would use no more entries as before, in fact we can reuse the EEPROM locations.

Same approach could work for TX (here we have less data).

Later we can add additional points, which then can be used to fix the not fully linear curve. 

Any objections?

73
Danilo

**Comment by df8oe** [comment]
That's a good start.

We need 4 storage places for RX and 4 for TX:
RX_amp_low
RX_phase_low
RX_amp_high
RX_phase_high

TX_amp_low
TX_phase_low
TX_amp_high
TX_phase_high

Using this 4 values may be we are able to get a good approach over the whole range... As I see in configuration.h all values are 1 byte - so we can put phase AND gain in one "old" storage place.

**Comment by db4ple** [comment]
My intention was to reuse the existing 2x4 memory locations for RX and TX (since LSB and USB have (for obvious reasons) the more or less the same settings, I wanted to reuse the already used locations.
Normally it is a not such a brilliant idea to redefine the meaning of a setting but here I think we could get away with it since we use the data for almost the same purpose. 

In later iterations I think we would indeed use the upper byte for additional extra data points.
By this we put 2 gain values in the formerly only half used entries. 

Benefit of my proposal is that existing values can be simply used as is. In case both LSB and respective USB setting are similar (most likley) you will get a more or less constant value across the full frequency range (as today).  So even if you don't notice the additional capabilities it will still work almost as good as before.

**Comment by s53dz** [comment]
By using this simple approach of linear interpolation we kill two flies with one hit:
- improve IQ bal/ph compared to what we have,
- leave the data format as it is.

The fact that we can do more IQ balance on separate bands than we can use is a pity.
Of course with more precise approach we would gain more but it is also more complex to implement.

73 Bojan

**Comment by db4ple** [comment]
Hi Bojan,

the more precise approach will follow, I am pretty sure. But I would like to do it stepwise. Each step will be an improvement (hopefully). But maybe less is more in this case. Just let us get started.

73
Danilo

**Comment by s53dz** [comment]
If I was a bit unclear:
I support doing a two points linear interpolation.
I think we can gain a better rejection over frequency compared to now.

TNX 73 Bojan

**Comment by df8oe** [comment]
Let's start and test. We can only step forward if we try something...

**Comment by s53dz** [comment]
If I may, I would propose:

1. to use S-meter scale for testing the results of using lin. interpolation. The main purpose being to have all measurements on the same ruler. S meter scale is not big (may be not accurate-calibrated), however it allows for measurements which to my opinion are enough for this testing of method. As the difference can be quite high. The frequencies at which it would be measured don't matter as it can be easy drawn on a scatter chart. The chart would than show the deviation from the optimal balance at two points (measured and stored at the beginning).

2. If it turns out that for instance 10m band is deviating than we can move the upper point to say 12m instead to improve the method. 

73 Bojan

**Comment by s53dz** [comment]
Hi all, 
I just wanted to share my preliminary measurements after setting the two values for 80m and 10m and using interpolation.

The results show that the level of a mirror signal is from S3 to S5 compared to the worst case of S8 before.
The signal used was S9 +15dB = -58dBm.

Before: I have used the 20m values, where the mirror was faded out.
Now: I have faded out the mirror on 80m and 10m band.

In both cases the fade out was a deep one, down to S2 or less.
http://lea.hamradio.si/~s53dz/arhiv/x-razno/1-mcHF-70MHz/1-measurements/IQ_test1.pdf
Since the deep is really sharp I have observed also the influence of a temp drift.

73 Bojan

**Comment by s53dz** [comment]
And I added TX_IQ measurements still with a FW ver1.5.2!
http://lea.hamradio.si/~s53dz/arhiv/x-razno/1-mcHF-70MHz/1-measurements/IQ_test2.pdf

73 Bojan

**Comment by df8oe** [comment]
Very interesting to see phase and balance adjustments on different mcHFs. The influence of boards is enormous. But always linear iq calculation does a good job...

**Comment by s53dz** [comment]
I can comment only on my results (since I have not seen others):
- almost S5 (-97dBm) as a worst case on RX seems quite high to me
- -37dBc as a worst case on TX seems quite acceptable (1mW at 5W output)

73 Bojan S53DZ

**Comment by df8oe** [comment]
Other informations are scattered - not here on GitHub. We are starting collection of more informations now.

Linear interpolation is a big step again. Now we must see if there are other mathematical possibilities to get better proximation. Maybe there do exist some functions, between the user can select (depependent on mcHF PCB and/or modifications). Maybe there is no other possibility to add adjustment to much more datapoints (> 50). We don't know at the moment.

**Comment by PA3CCE** [comment]
My results of the new IQ calibration Rx.
mcHF version 0.5.
Latest daily Firmware 1.5.4 /BL:  1.0.0.
Potatoechips /4 in LO 
LO and Rx mixer metal screened.
Signal generator (FA-VA3) via 26dB attenuator on antenna connector, showing abt -40dBm and S9+30 on the S-meter.
All dBm values read from mcHF screen.

11/12/2016
Rx IQ calibration :
===================
 3.600.001 MHz 	Balance - 9
		                Phase   -12

28.100.001 MHz	Balance -11
                                Phase     3
Result :
--------
   USB	LSB    Signal/Mirror	
	
 3.600/ 3.624	-40/-80 dBm
 5.400/ 5.424	-38/-78 dBm
 7.100/ 7.124	-40/-80 dBm
10.100/10.124	-38/-79 dBm
14.100/14.124	-42/-84 dBm
18.100/18.124	-40/-81 dBm
21.100/21.124	-40/-80 dBm
28.100/28.124	-42/-81 dBm

73 de
Jos (PA3CCE)

**Comment by s53dz** [comment]
Hi Jos,

Thanks for sharing your results. It seems that interpolation works just fine.
However, I think I do not understand completely. Could you please comment.

- why the results on 80m and 10m (where the reference setting is done) are not better?
- have you tried to measure mirror at somewhat lower input level?

I have recalculated my results to dBc. It shows some 17dB difference.

73 Bojan

[IQ_test3_dBc.pdf](https://github.com/df8oe/mchf-github/files/644698/IQ_test3_dBc.pdf)

**Comment by s53dz** [comment]
Sorry,
My bad. Error at table copy.
This is the right attachment now.

Hi Jos,

Thanks for sharing your results. It seems that interpolation works just fine.
However, I think I do not understand completely. Could you please comment.

    why the results on 80m and 10m (where the reference setting is done) are not better?
    have you tried to measure mirror at somewhat lower input level?

I have recalculated my results to dBc. It shows some 17dB difference **at 80m end**.

73 Bojan S53DZ

[IQ_test3_dBc.pdf](https://github.com/df8oe/mchf-github/files/644706/IQ_test3_dBc.pdf)

**Comment by PA3CCE** [comment]
Hi Bojan

Well, that the results at the calibration points were not better than the others stroke me also..
The only explanatation I can give is that I did not do the dBc measurements immediatly after calibration.
I also observed that when I did a recalibration the settings differed slightly sometimes (temperature ???)

I will do the calibration again and measure the dBc directly afterwards to see if the values are better at that moment.
That 17 dB difference is at the lower frequencies, above they are quite similar.
Will let you know my findings..

73 de
Jos

**Comment by S52CQ** [comment]
I have followed instructions and done calibration both Rx & Tx - now I can see/hear a big dip in the signal when changing calibration.

Great job!
73 de Jure, S52CQ

**Comment by PA3CCE** [comment]
Hello Bojan,

I forgot to tell you abt the lower input level.
Yes I did, level was abt S7 at the time and the mirror rejection was abt the same 40dB's but instable caused by the noise. So I decided to increase the level considerably.

Jos,

**Comment by s53dz** [comment]
Jos,

Thanks a lot for your comments.
Yes, I too noticed the very sharp behavior of I/Q settings, so the influence of temperature drift is high here.
I have rechecked this Issue thread for whole and I can repeat what was already said: Nothing can replace the initial I/Q error coming from the HW! Although I am happy to see this important improvement in the mcHF life using a simple linear interpolation. Respect.

73 Bojan S53DZ

**Comment by PA3CCE** [comment]
Bojan

I just did some tests at 3.6MHz :

1)
I  put the 26dB attenuator after the frequency generator and measured -40dBm carrier on the mcHF.
Then I adjusted the mirror at minimum, that level was -78 dBm on the mcHF, so -38 dBc.
The gain setting was -7 and the phase setting  -12.

2) 
Now I exchanged the 26dB attenuator for one of 40 dB.
The mcHF showed -54dBm, so 14 dB's less that is excellent.
Then I re-adjusted the mirror rejection.
Now the gain nsetting came out at -12 and the phase setting stayed at -12
Conclusion :  gain setting is level dependent or compression @ -40dBm .
However, level of mirror was -91 dBm --> -37dBc (let's say it's the same as before).

Will continue tomorrow.

73 
Jos

**Comment by PA3CCE** [comment]
Should be compression @ -26 dB,. sorry

**Comment by PA3CCE** [comment]
Bojan,

The way I measured the mirror rejection  by reading the displayed values on the mcHF screen is NOT correct,  sorry.
I found that without signal on the antenna plug my noise level indicates -128 dBm on the mcHF screen.
When I connect the signal generator via the 40dB pad on the antenna connector I read -54 dBm on the frequency , but whern I tune off frequency the noise level now shows -93 dBm on the screen. 
I had set AGC off in the menu.
So that's the value I always will measure when I turned down the mirror completely  (abt  -40 dBc).
Anyhow, the reality will be better. Right ?
How did you measure the dBc,, on the audio?
 if so,  it should be a selective measurement on 1 kHz  to be correct..
Maybe better on 12kHz on a spectrum analyser, i will try that. 
Regards
Jos

**Comment by PA3CCE** [comment]
Bojan,

On 12kHz I only see the carrier on the spectrum analyser  obviously.
When I measure the audio out on my spectrum analyser using mcHF's smallest bandwith on SSB (1.3-2.3 kHz) the difference between max signal @ 1kHz and the noise is just a bit more than 40 dB.
So I never will be able to measure more than that abt. -40 dBc, though it will be more.
How did you manage, is your noise level better than mine ?

best regards
Jos

**Comment by s53dz** [comment]
Jos,

With a 50 Ohm dummy load (40dB+10dB attenuator) at the RF input and using a CW filter I have mcHF displaying an S1 unit. And shows around -130dBm. But at this measurement I have used the SSB filter instead since it is more convenient and also it is not important to go really into that deep. Then I have some S2 units. The dip itself is really too sharp and it quickly becomes irrelevant when you go off the freq. It would be informative to see the IQ unbalance when inside the band.

I forgot to mention that the TCXO temp at the time was nearly constant 45 deg C.
It is also helpful if the signal generator has PN at least -50dBc at 24kHz offset.

Measuring TX mirror is quite straight forward. Of course it would help to have a narrow BW filter to notch the carrier out but it is not feasible. I am happy with that.

**Added(s53DZ):**
My RX IQ measurements could be biased since I have trusted the S-scale of mcHF and then calculated dBm out of that. I have measured the input signal level with external power meter and it was -49,5dBm. mcHF showing -54dBm.
**Added 2(s53DZ):**
An example: S9+20dB = -53dBm, S2.7 = -110.8dBm, difference = 57.8dB
TX IQ measurements are ok.

73 Bojan S53DZ

**Comment by DG9BFC** [comment]
hello jos

this is how i would (try to)  measure that

test transmitter set to rx qrg ...

crank up till you see an s2 or so

now tune test transmitter to mirror

crank up till you again see s2 (or what you had with first setting) 
(added level is mirror surpression)

retune test tx to rx qrg ... (now you will see s9+ something) ...

no ... do not use that readout!!!! (cause you never know how accurate s 
metre is!!!)

add attenuator till you again see s2

added damping is your mirror surpression

dg9bfc sigi




Am 12.12.2016 um 14:32 schrieb PA3CCE:

**Comment by PA3CCE** [comment]
Bojan,

Thanks for the reply.
I now understand you first measure the input level with an external power meter and then calculate the mirror level from the S reading on the mcHF.

What I did is:  

1) Reading the dBm value on the mcHF at the signal frequency and subtracting the dBm reading at        the mirror frequency.

2) I observed the audio spectrum on a spectrum analyser and measured the peak value of the 1 kHz signal to be 40dB over the noise .
Since the mirror was turned down into that noise I only could say  "mirror suppression is > 40 dB".

So both methods gave the same result > 40dB, but can't say how much due to the noise.

**Comment by PA3CCE** [comment]
Hi Sigi,

Thanks for your suggestion.
However, I am satusfied with the result > 40dB, knowing at all frequencies it is ( probably mostly far) better than that.
Anyway a big improvement compared to the old situation.

Thanks agn es
73 de
Jos

**Comment by dalerobins** [comment]
Hi all,

I'm after some advice regarding the IQ Phase/Balance.
I recently built my mcHF and am using the amazing firmware by DF8OE (Latest version)

One thing that is bothering me is on 10m to completely remove the "tone" my IQ Phase is around +124, this seems quite high to me as I don't have much adjustment left.

Frequency correction was on -12Khz to do the adjustments.

My board is a V0.5 from Chris, no modifications at all.

Is this normal or is something wrong, I am worried maybe I have not done something correctly, will having my phase so high cause any issues? - If something is wrong what do I need to do to correct it.

Many thanks, and keep up the great work!

de 2W0ODS IO81

**Comment by df8oe** [comment]
I think your 74xxx74 is at the border of its working capabilities and has very much phase differ at higher frequencies.

We have investigated that usage of 7474 manufactured by "potatoe chips" is working much proper. It is capable to work up to 1GHz so 120MHz (30MHz * 4) is not a problem at all.

Unfortunately Chris has redesigned 0.5 unluckily:
Vdd of 7474 in 0.5 is now 5V - but potatoe chip can only be used at 3.3V. So you have to modify Vdd for divider if you swap to potatoe chip.

Easier way is to try another manufacturer of 7474 to get a faster one...

73 de
Andreas

**Comment by dalerobins** [comment]
Andreas,

Thank you so much for a quick reply, was not expecting to hear anything today. 
Merry Christmas to you.

I see, so it is not anything I have done when winding toroid or transformer. That is a relief.

I will look at the Potatoe Chips (That is a funny name), or a faster 7474. 

Will everything work okay on my mcHF as it is, or is it causing a problem having the phase so high on 10m, should I not worry about it?

Love your firmware, it is working very well here.  Noticed a couple of couple of minor bugs (display corruption when changing S-Meter mode and you do not have the dBm/Hz display enabled, enabling it removes the corruption. Also waterfall you cannot see the > after magnify)

**Comment by df8oe** [comment]
H Dale,

Merry Christmas, too! I am spending some mnutes on GitHub every day - Christmas time included :)

There will be not a big problem. Your adjustings are within possible range. But I guess linear calculation for all other bands will not be as exact as they can be if the differences between 80m and 10m in settings are high. Phase error may be non-linear increasing at higher frequencies so it will be better if you reduce hardware phase errors as good as you can do it...

Take a look at the datasheet from semipotato: [http://www.potatosemi.com/datasheet/PO74G74A.pdf](url)

73 de
Andreas

**Comment by dalerobins** [comment]
Thank you again Andreas,

One thing I forgot to mention and I hope I am not causing any confusion here, the IQ Phase I adjusted on 10m is for TX, that value is at 124, I have not done the Rx ones yet but will do those today.

Also does anyone else have higher SWR readings on higher frequencies with a known good dummy load, I it registers 2-3 on the SWR meter.  I hope this is not causing the radio to put out a lower output power.

**Comment by df8oe** [comment]
Hi Dale,

output power is not reduced at higher SWRs.

There is a design fault for tandem couplers. Please take a look at our German Project Group page for modification RF-04-H-023. Sorry it's available in German actually :/ The toroid windings (many windings) which lead to PA output must be removed and connected to BNC side (it's connected to the wrong "side"). If you correct this readings will be more accurate.

RX and TX IQ settings are not cross-dependent - so no care to first do only one setiing.

73 de
Andreas

**Comment by dalerobins** [comment]
Is this still an issue with v0.5 board, I was ignoring any "04" mods as I
thought they were not applicable.
I will go back and look at all the modifications on your Wiki again.

Just wanted to check with the IQ TX too see if it is still 7474 for Tx and
if +124 is still okay for Tx.

I can use google translate, if not my close friend who lives in Heidelberg
can help me.

I love Germany it is one of my favorite countries, when I worked for HP I
spent a lot of time there, mainly in Boeblingen.
My friend moved from the UK to Germany and would not come back here.

**Comment by df8oe** [comment]
Hi Dale,

124 is "the edge". But if you cann suppress mirror deeply it will work. For better results in calculated settings try faster flip flop 7474...

We do not have completed mod comparison 0.4/0.5 because none of the fw developers own 0.5 boards. Because Chris removed live debugging headers on 0.5 there is no mitivation to change to this revision.... But I think many modifications are NOT implemented in 0.5 boards... Unluckily you have to test this by yourself :/

Good luck - 73
de Andreas
